src/main/kotlin/day16/data/MockImageDataSource.kt

```
@@ -0,0 +1,15 @@
+import kotlinx.coroutines.delay
+import kotlin.io.path.Path
+
+class MockImageDataSource(private val delayAmount: Long) : ImageDataSource {
+    val mockImage = Path("src/main/kotlin/day16/data/img.png").toFile()
+    override suspend fun fetchImage(url: String): ByteArray {
+        delay(delayAmount)
+        return mockImage.readBytes()
+    }
+
+    override suspend fun saveImage(bytes: ByteArray, path: String) {
+    }
```

detekt (1.23.7)

[warning] This empty block of code can be removed.

(detekt.empty-blocks.EmptyFunctionBlock)